src/app/clusters/bindings/BindingManager.cpp

```diff
@@ -176,6 +178,19 @@ void BindingManager::HandleDeviceConnectionFailure(PeerId peerId, CHIP_ERROR err
 {
     // Simply release the entry, the connection will be re-established as needed.
     ChipLogError(AppServer, "Failed to establish connection to node 0x" ChipLogFormatX64, ChipLogValueX64(peerId.GetNodeId()));
+
+    for (PendingNotificationEntry pendingNotification : mPendingNotificationMap)
+    {
+        EmberBindingTableEntry entry = BindingTable::GetInstance().GetAt(pendingNotification.mBindingEntryId);
+
+        PeerId peer = PeerIdForNode(mInitParams.mFabricTable, entry.fabricIndex, entry.nodeId);
+
+        if (peerId == peer)
+        {
+            mBoundDeviceContextReleaseHandler(pendingNotification.mContext);
```

Comment:
But this does not clear the pending notification, right?  Doesn't it just stick around with a dangling pointer, possibly to be dereferenced if in the future we manage to establish a connection to this peer?

Reply:
I think the last commit also addresses it, as now it will not be possible to delete context when any pending notification left.